src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp

```diff
@@ -302,6 +302,86 @@ class GroupQueryAttentionDecomposition : public ov::pass::MatcherPass {
     }
 };
 
+class GemmaSlidingMask : public ov::pass::MatcherPass {
+public:
+    OPENVINO_MATCHER_PASS_RTTI("npuw::LLMCompiledModel::GemmaSlidingMask");
+
+    struct Result {
+        Result() = default;
+
+        bool found = false;
+        int32_t window_size = 0;
+        std::shared_ptr<ov::op::v0::Parameter> mask_input;
+    };
+
+    explicit GemmaSlidingMask(Result* result) {
+        // Searching for gemma sliding mask pattern and replace it's output
+        // with Paramater of the same size and type.
+        /*                                                            -\
+          range -> unsqueeze -> unsqueeze -> unsqueeze1 -> convert -\/  => LessEqual -\
+                                                             \      /\-/               \
+                                                        /-----\----/                    =>BWAnd_res
+                                                       /       \                       /
+          range -> unsqueeze -> unsqueeze -> unsqueeze2 -> add -=> Greater -> BWAnd --/
+                                 const (-window_size) ----/
+        */
+        // Basically this subgrapgh is doing the following check:
+        // unsqueze2 - window_size < unsqueeze1 <= unsqueeze2
```

Comment:
Can we please add a note about `i < j` representation, when `i` is a row and `j` as column, I like that in your explanation!

Reply:
Done